deploy/crds/apps.3scale.net_apimanagerrestores_crd.yaml

```diff
@@ -0,0 +1,122 @@
+              properties:
+                persistentVolumeClaim:
+                  properties:
+                    claimSource:
+                      description: PersistentVolumeClaimVolumeSource references the
+                        user's PVC in the same namespace. This volume finds the bound
+                        PV and mounts that volume for the pod. A PersistentVolumeClaimVolumeSource
+                        is, essentially, a wrapper around another type of volume that
+                        is owned by someone else (the system).
+                      properties:
+                        claimName:
+                          description: 'ClaimName is the name of a PersistentVolumeClaim
+                            in the same namespace as the pod using this volume. More
+                            info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#persistentvolumeclaims'
+                          type: string
+                        readOnly:
+                          description: Will force the ReadOnly setting in VolumeMounts.
+                            Default false.
+                          type: boolean
+                      required:
+                      - claimName
+                      type: object
+                  required:
+                  - claimSource
+                  type: object
+                simpleStorageService:
+                  properties:
+                    bucket:
+                      type: string
+                    credentialsSecretRef:
+                      description: LocalObjectReference contains enough information
+                        to let you locate the referenced object inside the same namespace.
+                      properties:
+                        name:
+                          description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+                            TODO: Add other useful fields. apiVersion, kind, uid?'
+                          type: string
+                      type: object
+                    endpoint:
+                      type: string
+                    forcePathStyle:
+                      type: boolean
+                    path:
+                      type: string
+                    region:
+                      type: string
+                  required:
+                  - bucket
+                  - credentialsSecretRef
+                  type: object
+              type: object
+          required:
+          - restoreSource
+          type: object
+        status:
+          description: APIManagerRestoreStatus defines the observed state of APIManagerRestore
+          properties:
+            apiManagerBackupSharedInSecret:
+              type: boolean
+            apiManagerBackupSharedInSecretCleanup:
+              type: boolean
+            apiManagerRestoreStepFinished:
+              type: boolean
+            completed:
```

Comment:
The thing with conditions is that they can only take the value "true/false", not an "enum" of states.
Conditions are also a little bit harder to manipulate (searching, adding, removing elements etc...) so I went the simpler way.

Additionally, we could end up wanting more complex information than just a state, that's why I was thinking whether it would be a good idea to structure more the status, like:

.status.pvcDestination.configmapsAndSecrets.BackupFinished and other kind of potential information like .status.pvcDestination.logs

etc...

How would we fit the potential new information and lack of inherent structure of conditions?

Reply:
Followed a different approach based on K8s objects direct state observation and auxiliary bools in case they were needed